lib/broccoli/ember-app.js

```diff
@@ -1176,8 +1191,19 @@ EmberApp.prototype.testFiles = function(coreTestTree) {
 
   var emberCLITree = this._processedEmberCLITree();
 
-  var testJs = this.concatFiles(mergeTrees([external, coreTestTree]), {
-    inputFiles: this.legacyTestFilesToAppend,
+  var coreExt = mergeTrees([external, coreTestTree])
+
+  coreExt = stewMap(coreExt, function(content, relativePath){
+    if (legacyTestFilesToAppend.indexOf(relativePath) > -1) {
```

Comment:
i would recommend instead using `header` and `footer` slots

```js
var node = concat(node, {
  outputFile: '/output.js',
  header: ";(function() {",
  inputFiles: ['**/*']
  footer: "}());"
});
```

that being said, I believe this is would require a per file header/footer.

maybe something like `map` should be added to sourceMap concat, if map returns an `array` each chunk should be source-map aware concated?

```js
var node = concat(node, {
  outputFile: '/output.js',
  inputFiles: ['**/*'],

  map(content) {
    return [
      'perf file header',
      content,
      'per file footer'
    ];
  }
});
```